application/services/event_listener.py

```diff
@@ -0,0 +1,57 @@
+from config import EVENT_FETCH_LIMIT
+from common.blockchain_util import BlockChainUtil
+from infrastructure.repositories.contract_repository import ContractRepository
+from infrastructure.repositories.event_marker_repository import EventMarkerRepository
+from infrastructure.repositories.event_repository import EventRepository
+
+blockchain_util = BlockChainUtil()
+
+contract_repo = ContractRepository()
+event_marker_repo = EventMarkerRepository()
+event_repo = EventRepository()
+
+
+class EventListener:
+    def __init__(self, contract_name):
+        blockchain_util.test_connection()
```

Comment:
rename the function name `test_connection`

Reply:
Test_connection tests the connection and resets if the connection is down.